dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIRequestSettingsConverter.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI;
+
+/// <summary>
+/// JSON converter for <see cref="OpenAIRequestSettingsConverter"/>
+/// </summary>
+public class OpenAIRequestSettingsConverter : JsonConverter<OpenAIRequestSettings>
```

Comment:
This helps -- we could do similar with ooga booga too, right?

Reply:
Strictly we don't even need to include this for this option (it's here because I created options 2,3 from the option 1 branch). This is needed in option 1&2 to convert any anonymous class using pascal case e.g. If I set `requestSettings: new { MaxTokens = 128 }` in code. We want to use the pascal case because it's C# code. But for option 3 we cannot support that option so this converter can be removed.

Reply:
> For Oobabooga it has defined it's own AI request settings class and it just needs to set the appropriate JSON property name attributes and things will work.

Okay cool, then I'd propose that connector does the conversion from `StopSequences` to `stopping_sequences` .

Reply:
We want to avoid one connector needing to know the settings another connector supports or all connectors needing to know how to adapt OpenAI specific request settings to the pattern they support. If OpenAI bring out a new request setting then only it's connector should need to change.